src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/TrustsDataController.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.model.corporatetrustee.PrivateCorporateTrusteeListApi;
+import uk.gov.companieshouse.api.model.utils.Hashable;
+import uk.gov.companieshouse.api.model.utils.PrivateDataList;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.overseasentitiesapi.service.OverseasEntitiesService;
+import uk.gov.companieshouse.overseasentitiesapi.service.PrivateDataRetrievalService;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+import uk.gov.companieshouse.overseasentitiesapi.utils.Constants;
+import uk.gov.companieshouse.overseasentitiesapi.utils.HashHelper;
+
+@RestController
+@RequestMapping("/private/transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/trusts")
+public class TrustsDataController {
+
+    PrivateDataRetrievalService privateDataRetrievalService;
+    OverseasEntitiesService overseasEntitiesService;
+
+    @Value("${FEATURE_FLAG_ENABLE_ROE_UPDATE_24112022:false}")
+    private boolean isRoeUpdateEnabled;
+
+    @Value("${PUBLIC_API_IDENTITY_HASH_SALT}")
+    private String salt;
+    private HashHelper hashHelper = new HashHelper(salt);
+
+    private Map<String,Object> logMap;
+
+
+
+    @Autowired
+    public TrustsDataController(
+            final PrivateDataRetrievalService privateDataRetrievalService,
+            final OverseasEntitiesService overseasEntitiesService
+    ) {
+        this.privateDataRetrievalService = privateDataRetrievalService;
+        this.overseasEntitiesService = overseasEntitiesService;
+
+        this.privateDataRetrievalService.setHashHelper(hashHelper);
+    }
+
+
+    private Map makeLogMap(
+            String transactionId,
+            String overseasEntityId
+    ){
+        return Map.of(
+                Constants.OVERSEAS_ENTITY_ID_KEY, overseasEntityId,
+                Constants.TRANSACTION_ID_KEY, transactionId
+        );
+    }
+
+
+    @GetMapping("/{trust_id}/corporate-trustees")
+    public ResponseEntity<PrivateCorporateTrusteeListApi> getCorporateTrusts(
+            @PathVariable(Constants.TRANSACTION_ID_KEY) String transactionId,
+            @PathVariable(Constants.OVERSEAS_ENTITY_ID_KEY) String overseasEntityId,
+            @PathVariable(Constants.TRUST_ID) String trustId,
+            @RequestHeader(value = Constants.ERIC_REQUEST_ID_KEY) String requestId)
+            throws ServiceException {
+
+        logMap = makeLogMap(transactionId, overseasEntityId);
+        String companyNo = getCompanyNumber(overseasEntityId,requestId);
+        if(companyNo == null){
+            return ResponseEntity.notFound().build();
+        }
+        Callable<PrivateCorporateTrusteeListApi> function = () -> privateDataRetrievalService.getCorporateTrustees(trustId, companyNo);
+        return checkSubmissionDto(function, requestId, "corporate trustee");
```

Comment:
Maybe move the arrow function into this method call to tidy things up even more

Reply:
That would destroy the point of `checkSubmissionDto` as it wouldn't be able to be reused by the other methods

Reply:
Apologies i mean pass it in as a parameter like this:
        `return retrievePrivateTrustData(() -> privateDataRetrievalService.getCorporateTrustees(
                trustId, companyNo), requestId, "corporate trustee");`